src/safeds_runner/server/_memoization_map.py

```diff
@@ -1,16 +1,266 @@
+    "safeds.ml.classical.classification._support_vector_machine.SupportVectorMachine",
+    "safeds.ml.classical.regression._ada_boost.AdaBoost",
+    "safeds.ml.classical.regression._decision_tree.DecisionTree",
+    "safeds.ml.classical.regression._elastic_net_regression.ElasticNetRegression",
+    "safeds.ml.classical.regression._gradient_boosting.GradientBoosting",
+    "safeds.ml.classical.regression._k_nearest_neighbors.KNearestNeighbors",
+    "safeds.ml.classical.regression._lasso_regression.LassoRegression",
+    "safeds.ml.classical.regression._linear_regression.LinearRegression",
+    "safeds.ml.classical.regression._random_forest.RandomForest",
+    "safeds.ml.classical.regression._regressor.Regressor",
+    "safeds.ml.classical.regression._ridge_regression.RidgeRegression",
+    "safeds.ml.classical.regression._support_vector_machine.SupportVectorMachine",
+])
+
+
+def _is_explicit_identity_class(value: Any) -> bool:
+    """
+    Check, whether the provided value is whitelisted, by comparing the module name and qualified classname to assign an explicit identity.
+
+    Parameters
+    ----------
+    value
+        Object to check, if allowed to receive an explicit identity.
+
+    Returns
+    -------
+    result
+        True, if the object can be assigned an explicit identity, otherwise false.
+    """
+    return hasattr(value, "__class__") and (value.__class__.__module__ + "." + value.__class__.__qualname__) in explicit_identity_classes
+
+
+def _has_explicit_identity(value: Any) -> bool:
+    """
+    Check, whether an explicit identity was assigned to the provided object.
+
+    Parameters
+    ----------
+    value
+        Object to check
+
+    Returns
+    -------
+    result
+        Whether the object has been assigned an explicit identity.
+    """
+    return hasattr(value, "__ex_id__")
+
+
+def _has_explicit_identity_memory(value: Any) -> bool:
+    """
+    Check, whether a shared memory location was assigned to the provided object.
+
+    Parameters
+    ----------
+    value
+        Object to check
+
+    Returns
+    -------
+    result
+        Whether the object has been assigned shared memory location.
+    """
+    return hasattr(value, "__ex_id_mem__")
+
+
+def _set_new_explicit_identity_deterministic_hash(value: Any) -> None:
+    """
+    Assign a new explicit identity to the provided object, and assign a deterministic hash.
+
+    Parameters
+    ----------
+    value
+        Object to assign an explicit identity and a deterministic hash to
+    """
+    value.__ex_id__ = uuid.uuid4()
+    value.__ex_hash__ = hash(value)
+
+
+def _set_new_explicit_memory(value: Any, memory: SharedMemory) -> None:
+    """
+    Assign a shared memory location to the provided object.
+
+    Parameters
+    ----------
+    value
+        Object to assign a shared memory location to
+    """
+    value.__ex_id_mem__ = memory
+
+
+@dataclass(frozen=True)
+class ExplicitIdentityWrapper:
```

Comment:
This `ExplicitIdentityWrapper` is unused for now, but can be used as a fallback, if some classes don't provide a possibility to calculate a deterministic hash